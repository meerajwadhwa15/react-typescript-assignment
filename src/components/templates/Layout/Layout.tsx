import React, { FunctionComponent, ReactNode } from 'react'
import './style.css'

export type Props = {
  children: ReactNode
}

const Layout: FunctionComponent<Props> = ({ children }) => (
  <div className="container">{children}</div>
)

export default Layout
