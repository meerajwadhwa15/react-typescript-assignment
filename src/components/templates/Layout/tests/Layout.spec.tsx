import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Layout, { Props } from '../Layout'

const props: Props = {
  children: 'test1',
}

describe('Layout Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(<Layout {...props} />)
  })
})

describe('Layout Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(<Layout {...props} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
