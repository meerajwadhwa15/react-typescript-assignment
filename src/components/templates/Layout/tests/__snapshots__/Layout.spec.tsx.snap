// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snapshot testing Matches the snapshot 1`] = `
<div
  className="container"
>
  test1
</div>
`;
